lta/log_format.py

```diff
@@ -0,0 +1,36 @@
+# log_format.py
+
+from datetime import datetime
+from logging import Formatter
+import json
+from logging import LogRecord
+from typing import Optional
+
+
+class StructuredFormatter(Formatter):
+    def __init__(self,
+                 component_name: Optional[str] = None,
+                 component_type: Optional[str] = None,
+                 ndjson: bool = True) -> None:
+        self.component_name = component_name
+        self.component_type = component_type
+        self.indent = None if ndjson else 4
+        self.separators = (',', ':') if ndjson else (', ', ': ')
+        super(StructuredFormatter, self).__init__()
+
+    def format(self, record: LogRecord) -> str:
+        # ensure our log message has an ISO 8601 timestamp
+        data = {
+            'timestamp': datetime.utcnow().isoformat()
+        }
+        # copy everything provided to us in the LogRecord object
+        for key in record.__dict__:
+            data[key] = record.__dict__[key]
```

Comment:
Directly using `__` methods is generally a bad idea, as they're supposed to be private.  I think the `vars()` function does what you want here.